modules/tensor_mechanics/src/bcs/PressureTM.C

```diff
@@ -0,0 +1,61 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "PressureTM.h"
+#include "Function.h"
+#include "MooseError.h"
+
+template<>
+InputParameters validParams<PressureTM>()
+{
+  InputParameters params = validParams<IntegratedBC>();
+  params.addClassDescription("Applies a pressure on a given boundary in a given direction");
+  params.addRequiredParam<int>("component", "The component for the PressureTM");
+  params.addParam<Real>("factor", 1.0, "The factor to use in computing the pressure");
+  params.addParam<FunctionName>("function", "The function that describes the pressure");
+  params.addParam<PostprocessorName>("postprocessor", "Postprocessor that will supply the pressure value");
+  params.set<bool>("use_displaced_mesh") = true;
+  return params;
+}
+
+PressureTM::PressureTM(const std::string & name, InputParameters parameters)
+  :IntegratedBC(name, parameters),
+   _component(getParam<int>("component")),
+   _factor(getParam<Real>("factor")),
+   _function( isParamValid("function") ? &getFunction("function") : NULL ),
+   _postprocessor( isParamValid("postprocessor") ? &getPostprocessorValue("postprocessor") : NULL )
+{
+
+  if (_component < 0 || _component > 2)
+  {
+    std::stringstream errMsg;
+    errMsg << "Invalid component given for "
+           << name
+           << ": "
+           << _component
+           << "." << std::endl;
+
+    mooseError( errMsg.str() );
```

Comment:
We typically just assemble the string inside of the MOOSE error instead of building a stringstream object first. mooseError() is a macro so you can use `<<` inside of the parenthesis. Not a huge deal.